README.md

```diff
@@ -1476,6 +1476,28 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
 
+* <a id='generic-extensions'></a>(<a href='#generic-extensions'>link</a>) When extending generic types, prefer using generic bracket syntax (or sugared syntax for applicable standard library types) instead of generic type constraints. [![SwiftFormat: genericExtensions](https://img.shields.io/badge/SwiftFormat-genericExtensions-7B0051.svg)](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#genericExtensions)
+
+  <details>
+
+  ```swift
+  // WRONG
+  extension Array where Element == Star { … }
+  extension Optional where Wrapped == Spaceship { … }
+  extension Dictionary where Key == Moon, Element == Planet { … }
+  extension Collection where Element == Universe { … }
+  extension StateStore where State == SpaceshipState, Action == Spaceship { … }
+
+  // RIGHT
+  extension [Star] { … }
+  extension Spaceship? { … }
+  extension [Moon: Planet] { … }
+  extension Collection<Universe> { … }
+  extension StateStore<SpaceshipState, SpaceshipAction> { … }
```

Comment:
Maybe something like `extension Array where Element: PlanetaryBody { }` (which could in theory be `extension Array<some PlanetaryBody>` , but that syntax isn't allowed)? Any other examples that you think it would be good to mention?

Reply:
Although maybe this isn't necessary if we word the rule in a way that has no exceptions https://github.com/airbnb/swift/pull/194#discussion_r970167833

Reply:
I think we should update the wording though I still like showing an example of when it's not possible, and we can say in a comment that the example is not a "bound generic type". This way we're helping a reader who may not be as deep in Swift to learn.